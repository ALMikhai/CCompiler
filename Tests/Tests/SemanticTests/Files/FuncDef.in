int foo(struct a {} a1, int b) {
    struct a a2;
    float c;
}
