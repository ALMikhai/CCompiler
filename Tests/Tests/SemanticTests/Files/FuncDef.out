\-TRANSLATIONUNIT
  \-FUNC
    |-DECLSPEC
    | |-INT
    |-INITDECLARATOR
    | \-DECLARATOR
    |   \-FUNC
    |     |-foo
    |     \-PARAMLIST
    |       |-PARAMDECLARATOR
    |       | |-DECLSPEC
    |       | | |-STRUCTSPEC
    |       | | | |-a
    |       | \-DECLARATOR
    |       |   \-a1
    |       \-PARAMDECLARATOR
    |         |-DECLSPEC
    |         | |-INT
    |         \-DECLARATOR
    |           \-b
    \-COMPOUNDSTAT
      |-DECLLIST
      | |-DECL
      | | |-DECLSPEC
      | | | |-STRUCTSPEC
      | | | | |-a
      | | | | \-EMPTY
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     \-DECLARATOR
      | |       \-a2
      | \-DECL
      |   |-DECLSPEC
      |   | |-FLOAT
      |   \-INITDECLARATORLIST
      |     \-INITDECLARATOR
      |       \-DECLARATOR
      |         \-c

Structs { }
Symbols {
	FUNC returning INT
	Arguments{
		STRUCT called a :: a1
		INT :: b
	}
	Structs {
		STRUCT called a
		Members { }
	}
	Symbols {
		STRUCT called a :: a1
		INT :: b
		STRUCT called a :: a2
		FLOAT :: c
	} :: foo
}
